torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -411,15 +411,28 @@ std::vector<Shape> compute_shape_std(
   return {Shape(self.scalar_type(), {})};
 }
 
-std::vector<Shape> compute_shape_embedding_dense_backward(
+std::vector<Shape> compute_shape_embedding_dense_backward_symint(
     const at::Tensor& grad_output,
     const at::Tensor& indices,
-    int64_t num_weights,
+    c10::SymInt num_weights,
     int64_t padding_idx,
     bool scale_grad_by_freq) {
   // Based on aten/src/ATen/native/Embedding.cpp::embedding_dense_backward_cpu.
+
+  if (num_weights.is_symbolic()) {
+      c10::SymIntNode symbolicIntNode = num_weights.toSymIntNodeImpl();
+      auto* lazySymIntNode =
+          dynamic_cast<torch::lazy::SymIntNodeImpl*>(symbolicIntNode.get());
+      TORCH_INTERNAL_ASSERT(lazySymIntNode);
+      auto size_node = lazySymIntNode->node_;
+      auto static_value =
+          std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node)
+              ->getStaticValue();
```

Comment:
This is so bad, if we're doing LTC symints for real this needs to be much shorter

Reply:
yea, it looks like it would be easy to make this a helper function at least, i will do that in my next version of this